nodebuilder/header/service.go

```diff
@@ -17,12 +17,20 @@ import (
 type Service struct {
 	ex libhead.Exchange[*header.ExtendedHeader]
 
-	syncer    *sync.Syncer[*header.ExtendedHeader]
+	syncer    syncer
 	sub       libhead.Subscriber[*header.ExtendedHeader]
 	p2pServer *p2p.ExchangeServer[*header.ExtendedHeader]
 	store     libhead.Store[*header.ExtendedHeader]
 }
 
+// syncer bare minimum Syncer interface for testing
+type syncer interface {
```

Comment:
can we define this elsewhere then?

Reply:
Actually, this is fine.

Reply:
It's meant to be a localized private interface with no external usage.